### DISCUSSION TITLE
is the anyway to disabled the polyfills?

### DISCUSSION THREAD

**Discussion by lili21** [vercel/next.js#20354] [Help] [answered]
I'm using services like polyfill.io. no reason to bundle the polyfill into the code.

**Comment by jamesmosier** [comment, marked as answer]
There is no way to disable Next's built-in polyfills. The polyfills Next provides are only loaded in legacy browsers that support the [`nomodule` script attribute](https://stackoverflow.com/a/45947601/1554860).

**Reply by zephyratech** [reply]
We've been facing this limitation while optimizing our Next.js application for Lighthouse and Core Web Vitals.

The legacy JS bundle generated by Next's built-in polyfills still appears during performance analysis, even though the application targets modern browsers. Since these polyfills can't be disabled, they continue to contribute to JavaScript size and occasionally affect performance metrics.

It would be great if Next.js provided an option to:

* Disable built-in legacy polyfills entirely.
* Configure the browser targets for polyfill generation.
* Opt into a modern-only build for applications that don't need legacy browser support.

This would give developers more control over bundle size and help achieve better performance scores for projects that only support evergreen browsers.
